design/open/oidc-support.md

```diff
@@ -0,0 +1,28 @@
+## OIDC support in lakeFS
+
+As suggested by @arielshaqed, we started researching the option to implement OIDC support as part of lakeFS.
+
+In such a solution, the person who configures lakeFS will supply the parameters required for implementing OIDC, including the provider's domain, the client ID and secret, and a callback URL. Once activated, the standard (non-OIDC) cookie based lakeFS authentication will not be available. API access will still use credentials managed internally like today.
+
+As a first step, authorization will still be managed internally by lakeFS.
+
+### Open questions
+
+#### What do we save in our database?
+What information about users should we save in the lakeFS database?
+
+The minimum here is to save the unique user ID generated by the OIDC provider (the "subject" claim).
+
+However, this may not be enough. 
+Since we only want authentication (and not authorization) to be handled by the external provider, we should be able to link between a user and roles/policies. It seems that to do so we need to save some more info (at least username/email). Once we also move authorizaiton outside, it may be easier to avoid saving any PII on the lakeFS side.
```

Comment:
regarding the assumption that we'll have to save more information locally (username/email), I'm not sure that's true.
we can use a hash on top of the mail or the oidc user id, we will need to make sure every function returning the username/email, will now have to query the oidc in order to return good results.
thoughts?

Reply:
I'm considering the case where I show a list of users, and need to display the email for all of them.
So you need to ask for information about users other than the one which is currently logged in. This is not part of the OIDC protocol.
Once users use a concrete provider, we should allow them to implement an `auth.Service` that uses the provider's API to fetch information about users. WDYT?

Reply:
I think I might be missing you.
if anyone would like to use OIDC, they'll have to implement `auth.Service` in order to have a complete functioning IAM system?
I'm not sure that's a reasonable ask, we should preserve the current features.
Can't we use a separate user for listing users?
A user that the admin will need to provide (along with configuring OIDC)